homeassistant/components/osoenergy/__init__.py

```diff
@@ -0,0 +1,140 @@
+"""Support for the OSO Energy devices and setrvices."""
+from functools import wraps
+import logging
+
+from aiohttp.web_exceptions import HTTPException
+from apyosoenergyapi import OSOEnergy
+from apyosoenergyapi.helper.osoenergy_exceptions import OSOEnergyReauthRequired
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_API_KEY, CONF_SCAN_INTERVAL
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed, ConfigEntryNotReady
+from homeassistant.helpers import aiohttp_client, config_validation as cv
+from homeassistant.helpers.dispatcher import (
+    async_dispatcher_connect,
+    async_dispatcher_send,
+)
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.typing import ConfigType
+
+from .const import DOMAIN, PLATFORM_LOOKUP, PLATFORMS
+
+_LOGGER = logging.getLogger(__name__)
+
+CONFIG_SCHEMA = vol.Schema(
+    vol.All(
+        cv.deprecated(DOMAIN),
+        {
+            DOMAIN: vol.Schema(
+                {
+                    vol.Required(CONF_API_KEY): cv.string,
+                    vol.Optional(CONF_SCAN_INTERVAL, default=15): cv.positive_int,
+                },
+            )
+        },
+    ),
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+    """OSO Energy configuration setup."""
+    hass.data[DOMAIN] = {}
+
+    if DOMAIN not in config:
+        return True
+
+    conf = config[DOMAIN]
+
+    if not hass.config_entries.async_entries(DOMAIN):
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN,
+                context={"source": config_entries.SOURCE_IMPORT},
+                data={
+                    CONF_API_KEY: conf[CONF_API_KEY],
+                },
+            )
+        )
+    return True
```

Comment:
Without YAML, this function should no longer be here.

Reply:
Removed the async_setup method

Reply:
@ludeeus this is needed to import the existing YAML configuration to a config entry, right?
So I think it shouldn't be removed. Or am I missing something?

Reply:
New integration so no.

Reply:
Ah, I see, then also the import flow and the test for the import flow should be removed imo.